classes/CronData.php

```diff
@@ -350,25 +355,74 @@ public static function getMultipleCol($question,$project_id,$multipleRecords,$st
             return array(0=>$percent,1=>$tooltip,2=>$showLegendexMultiple,3=>$array_colors,4=>$tooltipTextArray);
         }
     }
-
-     public static function getPercent($recordsTotal, $missing, $overall, $showLegend, $option){
-        if($recordsTotal == 0 || ($recordsTotal == $missing && $option != "multiple")){
-            //No responses
-            $percent = "-";
-            $showLegend = true;
-        }else if(($recordsTotal - $missing) < 5){
-            //Fewer than 5 responses
-            $percent = "x";
-            $showLegend = true;
-        }else if(($recordsTotal - $missing) < 20){
-            //Fewer than 20 responses
-            $percent = $overall." *";
-            $showLegend = true;
-        }else{
-            $percent = $overall;
-        }
-        return array(0=>$percent,1=>$showLegend);
-    }
+	
+	public static function getPercent($recordsTotal, $missing, $overall, $showLegend, $option){
+		if($recordsTotal == 0 || ($recordsTotal == $missing && $option != "multiple")){
+			//No responses
+			$percent = "-";
+			$showLegend = true;
+		}else if(($recordsTotal - $missing) < 5){
+			//Fewer than 5 responses
+			$percent = "x";
+			$showLegend = true;
+		}else if(($recordsTotal - $missing) < 20){
+			//Fewer than 20 responses
+			$percent = $overall." *";
+			$showLegend = true;
+		}else{
+			$percent = $overall;
+		}
+		return array(0=>$percent,1=>$showLegend);
+	}
+	
+	public static function getDoesNotApplyCount($filteredData,$fieldName,$project_id) {
+		$doesNotApplyCount = 0;
+		
+		$outcome_labels = self::$module->getChoiceLabels($fieldName, $project_id);
+		$topScoreMax = count($outcome_labels);
+		
+		## For survey questions with 5 choices, a 5 usually indicates a "Does not apply" answer
+		if($topScoreMax == 5) {
+			$doesNotApplyRecords = REDCapCalculations::mapFieldByRecord($filteredData,$fieldName,['5'],false);
+			$doesNotApplyCount = count($doesNotApplyRecords);
+		}
+		
+		return $doesNotApplyCount;
+	}
+	
+	public static function getTopScoreRecords($filteredData,$fieldName,$project_id) {
```

Comment:
Is this function really used? I see it also in R4Report

Reply:
No, there's a bunch of functions that I wrote when I was trying to re-work the whole program that I then stopped using. I left them in there just because I didn't want to remove any functions in case I'd used them.

Reply:
Should we just remove it then?

Reply:
I removed the unused functions on CronData and ProjectData. I wanted to leave the functions I had on R4Report as it was a partial re-write of the cron process that I just didn't have time to finish.

Reply:
Gotcha! I don't mind leaving some functions if you want to re-write them later or the sort but if some we are not going to use at all it's better to remove them to avoid confusion :)